src/rebuild/scan.c

```diff
@@ -636,26 +636,39 @@ placement_check(uuid_t co_uuid, vos_iter_entry_t *ent, void *data)
 	return rc;
 }
 
-struct rebuild_iter_arg {
-	ds_iter_cb_t	callback;
-	void		*arg;
-};
+static int
+rebuild_container_scan_cb(daos_handle_t ph, uuid_t co_uuid, void *data)
+{
+	struct rebuild_scan_arg *arg = data;
+	struct rebuild_tgt_pool_tracker *rpt = arg->rpt;
+	int			rc;
+
+	/* resync DTXs' status firstly. */
+	rc = dtx_resync(ph, rpt->rt_pool_uuid, co_uuid, rpt->rt_rebuild_ver,
+			true);
+	if (rc) {
+		D_ERROR(DF_UUID" dtx resync failed: rc %d\n",
+			DP_UUID(rpt->rt_pool_uuid), rc);
+		return rc;
+	}
+
+	return ds_cont_iter(ph, co_uuid, rebuild_obj_scan_cb, data,
```

Comment:
Iterate all objects in a container without yield? That'll hog CPU too long when iterating a large container. (all hardware poll will be blocked and RPCs will timeout).

I think the proper way is to improve vos_iterate() a bit to support container, so that iteration callback will be able to yield and do entry delete, and the general vos_iterate() code will do re-probe accordingly. What do you think?